app/code/Magento/Quote/Observer/SendInvoiceEmailObserver.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Quote\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\Order\Email\Container\InvoiceIdentity;
+use Magento\Sales\Model\Order\Email\Sender\InvoiceSender;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class responsive for sending invoice emails when order created through storefront.
+ */
+class SendInvoiceEmailObserver implements ObserverInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var InvoiceSender
+     */
+    private $invoiceSender;
+
+    /**
+     * @var InvoiceIdentity
+     */
+    private $invoiceIdentity;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param InvoiceSender $invoiceSender
+     * @param InvoiceIdentity $invoiceIdentity
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        InvoiceSender $invoiceSender,
+        InvoiceIdentity $invoiceIdentity
+    ) {
+        $this->logger = $logger;
+        $this->invoiceSender = $invoiceSender;
+        $this->invoiceIdentity = $invoiceIdentity;
+    }
+
+    /**
+     * Send invoice email if allowed.
+     *
+     * @param Observer $observer
+     *
+     * @return void
+     */
+    public function execute(Observer $observer)
```

Comment:
Please specify return type to method.
```suggestion
    public function execute(Observer $observer): void
```

Reply:
Not allowed to do that because declared interface method  ObserverInterface::execute() have no such return type hinting.